### PR TITLE
RIA-3412 pay and submit appeal

### PR DESCRIPTION
**Pull request by tarun-palisetty**
### JIRA link (if applicable) ###
[RIA-3412](https://tools.hmcts.net/jira/browse/RIA-3412)

### Change description ###
Pay and submit appeal journey

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java [hunk 1]
```diff
@@ -314,7 +314,10 @@
         return new NotificationHandler(
             (callbackStage, callback) ->
                 callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
-                    && callback.getEvent() == Event.SUBMIT_APPEAL
+                    && Arrays.asList(
+                            Event.SUBMIT_APPEAL,
+                            Event.PAY_AND_SUBMIT_APPEAL)
```
Comment (makowalski) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java:319]: this covers happy path only, we need to have error handling in follow up story

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java [hunk 2]
```diff
@@ -345,7 +348,10 @@
 
         return new NotificationHandler(
             (callbackStage, callback) -> callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
-                    && callback.getEvent() == Event.SUBMIT_APPEAL, notificationGenerators
+                    && Arrays.asList(
+                            Event.SUBMIT_APPEAL,
+                            Event.PAY_AND_SUBMIT_APPEAL)
```
Comment (makowalski) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java:353]: this covers happy path only, we need to have error handling in follow up story

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java (hunk 1 above)
Comment (Lee365) [reply src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java:319]: Noted

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java (hunk 2 above)
Comment (Lee365) [reply src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java:353]: Noted

**Comment by makowalski** [comment]
changes merged with RIA-3082 ticket

**Pull request closed without merging**